packages/hoppscotch-common/src/services/inspection/inspectors/response.inspector.ts

```diff
@@ -26,54 +31,163 @@ export class ResponseInspectorService extends Service implements Inspector {
 
   private readonly inspection = this.bind(InspectionService)
 
+  private inspectionResults = ref<InspectorResult[]>([])
+  private isLoadingAI = ref(false)
+
+  private async updateInspectionResults(results: InspectorResult[]) {
+    this.inspectionResults.value = [...results]
+    await nextTick()
+  }
+
   override onServiceInit() {
     this.inspection.registerInspector(this)
   }
 
   getInspections(
-    _req: Readonly<Ref<HoppRESTRequest | HoppRESTResponseOriginalRequest>>,
+    req: Readonly<Ref<HoppRESTRequest | HoppRESTResponseOriginalRequest>>,
     res: Readonly<Ref<HoppRESTResponse | null | undefined>>
   ) {
+    const { shouldEnableAIFeatures } = useAIExperiments()
+
+    // Watch for response changes to trigger inspection updates
+    watch(
+      [res, shouldEnableAIFeatures],
```

Comment:
You dont need to add the `shouldEnableAIFeatures` into the watcher. whenever the function is invoked via a response change, it will be getting the latest value.